test/unit/composition.test.js

```diff
@@ -0,0 +1,383 @@
+
+    const compositionEmpty = new Composition();
+    expect(compositionEmpty.prepend(...items)).to.deep.equal(new Composition(...items));
+
+    // *prepend* puts items towards the beginning
+    const composition = new Composition(' end');
+    expect(composition.prepend(...items).items).to.deep.equal(items.concat([' end']));
+
+    // *prepend* modifies instace without need to reassign
+    expect(composition.items).to.deep.equal(items.concat([' end']));
+
+    // *prepend* returns same object
+    expect(composition === composition.prepend(...items)).to.be.ok;
+  });
+
+  it('should have *clone* method', () => {
+    const items = ['foo ', new Slot(5), ' bar'];
+    const originalComposition = new Composition(...items);
+
+    // *clone* should return a different object
+    expect(originalComposition.clone() === originalComposition).to.not.be.ok;
+
+    // But the returned object must have the same items
+    expect(originalComposition.clone().items).to.deep.equal(items);
+  });
+
+  it('should have *from* method', () => {
+    const items = ['foo ', new Slot(5), ' bar'];
+    const composition = Composition.from(items);
+
+    expect(composition.items).to.deep.equal(['foo ', new Slot(5), ' bar']);
+
+    // *from* should create a *Composition* instance
+    expect(composition instanceof Composition).to.be.ok;
+  });
+
+});
+
+describe('Composition Group', () => {
+
+  it('accepts composition elements', () => {
+    expect(() => new CompositionGroup('foo ',  new Slot(5),
+      ' bar', new Composition())).to.not.throw();
+    // validation is done when using method toComposiiton()
+  });
+
+  it('should have *add* method', () => {
+    const compositions = ['foo ', new Slot(5), ' bar', new Composition()];
+
+    const groupEmpty = new CompositionGroup();
+    expect(groupEmpty.add(...compositions)).to.deep.equal(CompositionGroup.from(compositions));
+
+    // *add* puts compositions towards the end
+    const group = new CompositionGroup('start ');
+    expect(group.add(...compositions)).to.deep.equal(new CompositionGroup('start ', ...compositions));
+
+    // *add* modifies instace without need to reassign
+    expect(group).to.deep.equal(new CompositionGroup('start ', ...compositions));
+
+    // *add* returns same object
+    expect(group === group.add(...compositions)).to.be.ok;
+  });
+
+  it('should have *space* method', () => {
+    const compositions = ['foo ', new Slot(5), ' bar', new Composition()];
+
+    const group = CompositionGroup.from(compositions);
+    expect(group.space(', ').compositions).to.deep.equal(['foo ', ', ',
+      new Slot(5), ', ', ' bar', ', ', new Composition()]);
+
+    // *space* modifies instace without need to reassign
+    expect(group.compositions).to.deep.equal(['foo ', ', ',
+      new Slot(5), ', ', ' bar', ', ', new Composition()]);
+
+    // *space* returns same object
+    const group2 = CompositionGroup.from(compositions);
+    expect(group2 === group2.space(', ')).to.be.ok;
```

Comment:
Can we avoid using `.to.be.ok` assertions please?
They tend to hide diff's when they fail.